http_server.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import socket
+import email.utils
+import os
+
+
+def server_socket_function():
+    server_socket = socket.socket(socket.AF_INET,
+                                  socket.SOCK_STREAM,
+                                  socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+
+    try:
+        while True:
+            conn, addr = server_socket.accept()
+
+            recieve_total = ""
+            buffersize = 32
+            finished = False
+            while not finished:
+                recieve = conn.recv(buffersize)
+                if len(recieve) < buffersize:
+                    finished = True
+                recieve_total += recieve
+
+            if recieve_total:
+                try:
+                    uri = parse_request(recieve_total)
+                    info = resolve_uri(uri)
+                    response = response_ok(info)
+                except RequestError as error:
+                    response = response_error(error)
+                conn.sendall(response)
+            conn.close()
+    except KeyboardInterrupt:
+        server_socket.close()
+
+
+def parse_request(client_request):
+    mup_line = client_request.splitlines()[0]
+    mup = mup_line.split(' ')
+    uri = error_mup(mup)
+    return uri
+
+
+def error_mup(mup):
+    http_response_codes = {'405': 'Method Not Allowed',
+                           '505': 'HTTP Version Not Supported'}
+
```

Comment:
Constant values like this are often better defined at module scope.  That way you only define them once.  As it stands here, you re-generate this dict each time this function runs.  Not much of a cost, but over thousands of requests it would add up.